### PR TITLE
[Sync VIIc] Final alignment cleanup and name changes

### LINKED ISSUE #301: [Feature]: Reach agreement on temporal alignment method/argument names
### What would you like to see added to NeuroConv?

From discussions following #237, the usage of the word `timestamps` is a bit overloaded.

Specifically, `timestamps` ought to solely refer to the time vector of the associated `TimeSeries` of each interface, but it currently also applies to the time vectors used in `align_by_interpolation` which may be parsed from TTL or other digital signals.

Also, specific to `align_by_interpolation`, a description of what the function does in words could be "A digital signal is sent from a secondary system and received by the primary system to align the timing differences between the two clocks."

The keyword arguments for the function should be chosen to best describe the role of the inputs

Current: `unaligned_timestamps, aligned_timestamps`
Alternative #1: `ttl_sent_times, ttl_recieved times`
Alternative #2: `unaligned_times, aligned_times`
Alternative #3: `sent_times, recieved_times`
Alternative #3: `secondary_times, primary_times`

For alternative #2, keep in mind that the timing information does not need to be from TTL pulses - it could be inferred from an algorithm similar to how MICrONS does it - or it could be parsed from a non-TTL digital signal, such as audio events

### Is your feature request related to a problem?

_No response_

### Do you have any interest in helping implement the feature?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/neuroconv/blob/main/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/neuroconv/issues)?

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
fix #301 

Various cleanups around the repo to bring things up to the current standard for temporal alignment related methods and tests

Also adjusting the final naming conventions for the few methods that are simple setters as opposed to those that perform actual operations

On that note, I did notice that `align_starting_time` and `align_segment_starting_times` do actually perform operations on the underlying objects in the majority of cases so they aren't strictly setters; special cases include icephys, sorters without recordings attached, and sometimes video; but since it always has the possibility of being an operation as opposed to a strict setter 

As such it is only the `timestamps` notion that has setters and getters as methods referencing some type of underlying property

### CODE REVIEW

**Review by h-mayorquin — COMMENTED** [review]
I don't understand your comment about the sorter not setting the time.

https://github.com/catalystneuro/neuroconv/blob/bc258f225431ffc36044b4e800180b5c767badd3/src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py#L196-L204

To me,  that code above is setting the `start_time` of the interface.  Can you elaborate how you are thinking? (Maybe there is a better example so feel free to quote another piece of code if that's better).

**Comment by CodyCBakerPhD** [comment]
> To me, that code above is setting the start_time of the interface. Can you elaborate how you are thinking? (Maybe there is a better example so feel free to quote another piece of code if that's better).

If a recording is not attached, it does set the segment `._t_start` property

If a recording is attached, it modifies the timestamps in the exact same way as `align_starting_time` does

File: tests/test_on_data/test_behavior_interfaces.py [hunk 1]
```diff
@@ -18,7 +16,19 @@
     from setup_paths import BEHAVIOR_DATA_PATH, OUTPUT_PATH
 
 
-class TestDeepLabCutInterface(DataInterfaceTestMixin, TemporalAlignmentMixin, unittest.TestCase):
+class TestVideoInterface(VideoInterfaceMixin, unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (CodyCBakerPhD) [reply tests/test_on_data/test_behavior_interfaces.py:19]: Just style, to put DLC and SLEAP next to each other

Long-term probably best to split each up over a subfolder

**Comment by h-mayorquin** [comment]
> > To me, that code above is setting the start_time of the interface. Can you elaborate how you are thinking? (Maybe there is a better example so feel free to quote another piece of code if that's better).
> 
> If a recording is not attached, it does set the segment `._t_start` property
> 
> If a recording is attached, it modifies the timestamps in the exact same way as `align_starting_time` does

Yes, but I am not thinking it as a setter of the `t_start` property as that is a concept that only some interfaces have. I am thinking this as just a conenience method for setting the first timestamp. Whether this is implemented with a recording time vector, some internal timestamps, a subsegment t_start or a recording attached to a sorter is an implementation detail that should not be leaked in the interface. What matters is that there are some timestmaps associated with the interface and we can either set them all with set_timestamps or we can set the first one with `set_aligned_start_time` (which internally also shifts all the timestamps if that's the way they are represented).

In brief, is a setter of a time property, not of the underlying `segment._t_start` property.

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin This should be ready for final review now

**Review by h-mayorquin — APPROVED** [review]
Good.

**Pull request merged**